src/components/transport_manager/src/bluetooth/bluetooth_transport_adapter_win.cc

```diff
@@ -156,8 +146,17 @@ bool BluetoothTransportAdapter::Restore() {
         static_cast<uint8_t>(atoi(rfcomm_channel_record.c_str()));
       rfcomm_channels.push_back(rfcomm_channel);
     }
-    BluetoothDevice* bluetooth_device =
-      new BluetoothDevice(address, name.c_str(), rfcomm_channels);
+
+	BLUETOOTH_DEVICE_INFO  btdi = { 0 };
+	btdi.dwSize = sizeof(BLUETOOTH_DEVICE_INFO);
+	btdi.Address.ullLong = address.btAddr;
+	btdi.ulClassofDevice = 0;
+	btdi.fConnected = false;
+	btdi.fRemembered = false;
+	btdi.fAuthenticated = false;
+
+	BluetoothDevice* bluetooth_device = 
+			new BluetoothDevice(btdi, name.c_str(), rfcomm_channels);
```

Comment:
First where you create raw pointer and second when you create shared from this raw